nixos/modules/services/networking/3proxy.nix

```diff
@@ -0,0 +1,422 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  pkg = pkgs._3proxy;
+  cfg = config.services._3proxy;
+  optionalList = list: if list == [ ] then "*" else concatMapStringsSep "," toString list;
+in {
+  options.services._3proxy = {
+    enable = mkEnableOption "3proxy";
+    confFile = mkOption {
+      type = types.path;
+      example = "/var/lib/3proxy/3proxy.conf";
+      description = ''
+        Ignore all other 3proxy options and load configuration from this file.
+      '';
+    };
+    usersFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/var/lib/3proxy/3proxy.passwd";
+      description = ''
+        Load users and passwords from this file.
+
+        Example users file with plain-text passwords:
+
+        <para>
+          test1:CL:password1
+          test2:CL:password2
+        </para>
+
+        Example users file with md5-crypted passwords:
+
+        <para>
+          "test1:CR:$1$tFkisVd2$1GA8JXkRmTXdLDytM/i3a1"
+          "test2:CR:$1$rkpibm5J$Aq1.9VtYAn0JrqZ8M.1ME."
+        </para>
+
+        You can generate md5-crypted passwords via https://unix4lyfe.org/crypt/
```

Comment:
Maybe recommend `htpasswd` from `apacheHttpd` instead? Local tools sound better…

Reply:
Unfortunately ```htpasswd``` md5 hash is not compatible.

Reply:
That is surprising enough to mention as a comment then, in my opinion

Reply:
Done.